File: backend/app/models/guild.py
# backend/app/models/guild.py

from sqlalchemy import Column, String, ForeignKey, DateTime, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
import uuid

from app.db.base_class import Base


class Guild(Base):
    __tablename__ = "guilds"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String, nullable=False)
    icon_url = Column(String, nullable=True)

    owner_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    owner = relationship("User", back_populates="guilds")

    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), onupdate=func.now(), server_default=func.now(), nullable=False)

    channels = relationship("Channel", back_populates="guild", cascade="all, delete-orphan")

File: frontend/lib/core/models/guild.dart
// frontend/lib/core/models/guild.dart

class Guild {
  final String id;
  final String name;
  final String? iconUrl;
  final String ownerId;
  final DateTime createdAt;
  final DateTime updatedAt;

  Guild({
    required this.id,
    required this.name,
    this.iconUrl,
    required this.ownerId,
    required this.createdAt,
    required this.updatedAt
  });

  factory Guild.fromJson(Map<String, dynamic> json) {
    return Guild(
      id: json['id'],
      name: json['name'],
      iconUrl: json['icon_url'],
      ownerId: json['owner_id'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at'])
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'icon_url': iconUrl,
      'owner_id': ownerId,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String()
    };
  }
}
